core/base-service/loader.js

```diff
@@ -20,42 +21,72 @@ function loadServiceClasses(servicePaths) {
     servicePaths = glob.sync(path.join(serviceDir, '**', '*.service.js'))
   }
 
-  const serviceClasses = []
-  servicePaths.forEach(path => {
-    const module = require(path)
+  let serviceClasses = []
+  servicePaths.forEach(servicePath => {
+    const module = require(servicePath)
+
+    const theseServiceClasses = []
     if (
       !module ||
       (module.constructor === Array && module.length === 0) ||
       (module.constructor === Object && Object.keys(module).length === 0)
     ) {
       throw new InvalidService(
-        `Expected ${path} to export a service or a collection of services`
+        `Expected ${servicePath} to export a service or a collection of services`
       )
     } else if (module.prototype instanceof BaseService) {
-      serviceClasses.push(module)
+      theseServiceClasses.push(module)
     } else if (module.constructor === Array || module.constructor === Object) {
       for (const key in module) {
         const serviceClass = module[key]
         if (serviceClass.prototype instanceof BaseService) {
-          serviceClasses.push(serviceClass)
+          theseServiceClasses.push(serviceClass)
         } else {
           throw new InvalidService(
-            `Expected ${path} to export a service or a collection of services; one of them was ${serviceClass}`
+            `Expected ${servicePath} to export a service or a collection of services; one of them was ${serviceClass}`
           )
         }
       }
     } else {
       throw new InvalidService(
-        `Expected ${path} to export a service or a collection of services; got ${module}`
+        `Expected ${servicePath} to export a service or a collection of services; got ${module}`
       )
     }
+
+    // Decorate each service class with the directory that contains it.
+    theseServiceClasses.forEach(serviceClass => {
+      serviceClass.serviceFamily = servicePath
+        .replace(serviceDir, '')
+        .split(path.sep)[1]
+    })
+
+    serviceClasses = serviceClasses.concat(theseServiceClasses)
   })
 
   serviceClasses.forEach(ServiceClass => ServiceClass.validateDefinition())
 
   return serviceClasses
 }
 
+function assertNamesUnique(names, { message }) {
+  const duplicates = {}
+  Object.entries(countBy(names))
```

Comment:
this looks like fun 😄 just want to confirm that the validation of service name uniqueness only happens on server start up right?

Reply:
Only in the tests!

Reply:
👍